ui/StatusQ/src/sumaggregator.cpp

```diff
@@ -0,0 +1,45 @@
+#include "StatusQ/sumaggregator.h"
+
+void SumAggregator::calculateAggregation() {
+    auto tempValue = m_value;
+    m_value = 0;
+
+    // Check if m_model exists and role name is initialized
+    if (!m_model || m_roleName == "") {
+        if(tempValue != m_value) {
+            emit valueChanged();
+        }
+        return;
+    }
+
+    // Check if m_roleName is part of the roles of the model
+    QHash<int, QByteArray> roles = m_model->roleNames();
+    if (!roleExists()) {
+        qWarning() << "Provided role name does not exist in the current model";
+        emit valueChanged();
+        return;
+    }
+
+    // Do the aggregation
+    double total = 0.0;
+    int rows = m_model->rowCount();
+    int role = roles.key(m_roleName);
+
+    for (int row = 0; row < rows; ++row) {
+        QModelIndex index = m_model->index(row, 0);
+        QVariant value = m_model->data(index, role);
+
+        bool ok;
+        total += value.toDouble(&ok);
+
+        if (!ok)
+          qWarning() << "Unsupported type for given role (not convertible to double)";
+    }
+
+    m_value = total;
+
+    // Only emit signal if there are real changes:
+    if(tempValue != m_value) {
+        emit valueChanged();
+    }
+}
```

Comment:
One more thing - should simplify things even more:
```suggestion
QVariant SumAggregator::calculateAggregation() {
    // Check if m_model exists and role name is initialized
    if (!m_model || m_roleName.isEmpty())
        return 0.0;

    // Check if m_roleName is part of the roles of the model
    QHash<int, QByteArray> roles = m_model->roleNames();
    if (!roleExists()) {
        qWarning() << "Provided role name does not exist in the current model";
        return 0.0;
    }

    // Do the aggregation
    double total = 0.0;
    int rows = m_model->rowCount();
    int role = roles.key(m_roleName);

    for (int row = 0; row < rows; ++row) {
        QModelIndex index = m_model->index(row, 0);
        QVariant value = m_model->data(index, role);

        bool ok;
        total += value.toDouble(&ok);

        if (!ok)
          qWarning() << "Unsupported type for given role (not convertible to double)";
    }

    return total;
}
```
Changing the signature to return the value of calculated aggregation allows to move equality checks and signal emission to `Aggregator`.